src/main/java/pl/project13/maven/git/GitCommitIdMojo.java

```diff
@@ -198,9 +198,14 @@
    * Set this to {@code 'true'} to use native Git executable to fetch information about the repository.
    * It is in most cases faster but requires a git executable to be installed in system.
    * By default the plugin will use jGit implementation as a source of information about the repository.
+   *
+   * NOTE / WARNING:
+   * Do *NOT* set this property inside the configuration of your plugin, so it would be
+   * possible to override it from command line.
+   *
    * @since 2.1.9
    */
-  @Parameter(defaultValue = "false")
+  @Parameter(property = "maven.gitcommitid.nativegit", defaultValue = "false")
```

Comment:
Unfortunately I don't think this is working with maven and I'm still convinced that this is still utterly broken in maven.

The NOTE/WARNING you have introduced (copy and pasted from one of the other setting?) refers to https://github.com/git-commit-id/maven-git-commit-id-plugin/issues/315 where a similar problem was outlined for the `skip` property.

Essentially the problem here is that maven somehow does not respect the command-line option and will use the option configured in the pom (if available). However whenever a user has configured the setting in the pom the command-line option should overwrite that pom setting (IMHO that is at least how maven behaves). AFAIK this is not happening and is only possible when we need to introduce an additional property. 

This could look like something like this:
```java
  /** Set this to {@code 'true'} to use native Git executable to fetch information about the repository.
   * It is in most cases faster but requires a git executable to be installed in system.
   * By default the plugin will use jGit implementation as a source of information about the repository.
   *
   * @since 2.1.9
   */
  @Parameter(defaultValue = "false")
  boolean useNativeGit;


  /**
   * Set this to {@code 'true'} to use native Git executable to fetch information about the repository via commandline.
   * NOTE / WARNING:
   * Do *NOT* set this property inside the configuration of your plugin.
   * Please read
   * https://github.com/git-commit-id/maven-git-commit-id-plugin/issues/315
   * to find out why.
   * @since 3.0.2
   */
  @Parameter(property = "maven.gitcommitid.nativegit", defaultValue = "false")
  private boolean useNativeGitViaCommandLine;
```
Certainly whenever this plugin decided to run the native code it should run native whenever `useNativeGitViaCommandLine` or `useNativeGit` is true.

Why plugins are forced to do this this way is beyond me....and I would call it broken and not 'expected'...

Reply:
Thank you for advice - I've re-implemented it as you suggested. The new implementation checks, if the given property is set, and then use the value provided - this allows to overwrite setting even if `useNativeGit` is explicitly set to `true` in the project - for example, when I don't have Git installed on my machine.